packages/grid/_modules_/grid/hooks/features/sorting/useGridSorting.ts

```diff
@@ -211,25 +209,26 @@ export const useGridSorting = (apiRef: GridApiRef, rowsProp: GridRowsProp) => {
   );
 
   const sortColumn = React.useCallback(
-    (column: GridColDef, direction?: GridSortDirection) => {
+    (column: GridColDef, direction?: GridSortDirection, allowMultipleSorting?: boolean) => {
       if (!column.sortable) {
         return;
       }
       const sortItem = createSortItem(column, direction);
-      let sortModel: GridSortItem | GridSortItem[];
-      if (!allowMultipleSorting.current) {
+      let sortModel: GridSortItem[] | undefined;
+      if (!allowMultipleSorting || options.disableMultipleColumnsSorting) {
         sortModel = !sortItem ? [] : [sortItem];
       } else {
         sortModel = upsertSortModel(column.field, sortItem);
       }
       setSortModel(sortModel);
     },
-    [upsertSortModel, setSortModel, createSortItem],
+    [upsertSortModel, setSortModel, createSortItem, options.disableMultipleColumnsSorting],
   );
 
   const headerClickHandler = React.useCallback(
-    ({ colDef }: GridColumnHeaderParams) => {
-      sortColumn(colDef);
+    ({ colDef }: GridColumnHeaderParams, event: React.MouseEvent) => {
+      const isMultipleKeyPressed = event.shiftKey || event.metaKey || event.ctrlKey;
+      sortColumn(colDef, undefined, isMultipleKeyPressed);
```

Comment:
What if we use the enter key?

Reply:
It seems to work the same way https://codesandbox.io/s/charming-hopper-cpr1f?file=/src/App.js, https://deploy-preview-1422--material-ui-x.netlify.app/components/data-grid/sorting/#multi-column-sorting. What issue do you have in mind?

Reply:
hmm ok that's because enter Key or space key publish a column header click 🤔 
see useGridKeyboard line 215

Reply:
> hmm ok that's because enter Key or space key publish a column header click 🤔
> see useGridKeyboard line 215

so the event in the click would be of a wrong type if it is triggered from the keyboard.
I thing there should be more granular events here, like in `cellEditStart`

Reply:
Oh yeah, this is completely wrong:

https://github.com/mui-org/material-ui-x/blob/fa346f0fbe3d9b9eea9bb403fe4675f544d6abf9/packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts#L205-L210

Reply:
`GRID_COLUMN_HEADER_CLICK` should only be dispatched by a click event. It violates our rules when dispatched by other event. I'll add another listener just for the keyboard event.